app/models/user.rb

```diff
@@ -32,16 +32,20 @@ def media
   end
 
   def instagram_photos
-    return unless media['error'].nil?
+    _media = media
+    return unless _media['error'].nil?
 
+    media_count = _media['data'].size
     ig_photos = []
-    media['data'].each do |m|
+    media['data'].each_with_index do |m, i|
```

Comment:
@edance one thing to note for client-side loading animation is that I'm passing along media counts, not photo counts. Media count will always be <= photo count.